internal/kafka/internal/handlers/validation.go

```diff
@@ -407,3 +407,51 @@ func validateKafkaMachinePoolNodeCount(clusterPayload *public.EnterpriseOsdClust
 		return nil
 	}
 }
+
+func validateEnterpriseClusterEligibleForDeregistration(ctx context.Context, clusterID string, forceDeleteFlag bool, clusterService services.ClusterService) handlers.Validate {
+	return func() *errors.ServiceError {
+		claims, claimsErr := getClaims(ctx)
+
+		if claimsErr != nil {
+			return claimsErr
+		}
+
+		orgID, getOrgIdErr := claims.GetOrgId()
+
+		if getOrgIdErr != nil {
+			return errors.GeneralError(getOrgIdErr.Error())
+		}
+
+		if !claims.IsOrgAdmin() {
+			return errors.New(errors.ErrorUnauthorized, "non admin user not authorized to perform this action")
+		}
+
+		cluster, err := clusterService.FindClusterByID(clusterID)
+		if err != nil {
+			return err
+		}
+
+		if cluster != nil {
```

Comment:
we could inverse the condition here to 
```suggestion
		if cluster == nil {
```

and to and early return for the error case returned by line 455

Reply:
changed